src/components/userpanel/JumboStampSystem.tsx

```diff
@@ -0,0 +1,632 @@
+
+        stampIndividual[i].tier1 = Boolean(stamps[i]);
+        stampIndividual[i].tier2 = Boolean(stamps[i + 4]);
+        stampIndividual[i].tier3 = Boolean(stamps[i + 8]);
+      }
+    }
+  }, [stamps]);
+
+  return (
+    <JumboContainer>
+      <JumboTabsContainer>
+        <JumboTab>
+          <Tab>Meepo Market</Tab>
+        </JumboTab>
+        <JumboTab active>
+          <Tab active>Stamp System</Tab>
+        </JumboTab>
+      </JumboTabsContainer>
+
+      <JumboShadow active={showing}>
+        <JumboInnerBorder active={showing}>
+          <JumboInnerContainer>
+            {stampIndividual.map((stampIndividual: IndividualStampType) => (
+              <StampIndividual
+                key={stampIndividual.name}
+                stampIndividual={stampIndividual}
+              />
+            ))}
+
+            <MainContainer active={showing}>
+              <StampTopContainer active={showing}>
+                <StampTop />
+                <SeasonTab>
+                  <SeasonText>Season 1</SeasonText>
+                </SeasonTab>
+              </StampTopContainer>
+              <StampsRowContainer>
+                <StampsRow active={showing}>
+                  {stampIndividual.map((stamp: StampType) => (
+                    <JumboStamp
+                      key={stamp.id}
+                      stamp={stamp}
+                      show={() => (stamp.visible = true)}
+                    />
+                  ))}
+                </StampsRow>
+              </StampsRowContainer>
+              <RewardsContainer
+                onMouseEnter={() => {
+                  setHoverActive(true);
+                }}
+                onMouseLeave={() => setHoverActive(false)}
+              >
+                <ArrowDecorationDiv
+                  active={hoverActive}
+                  hide={showing}
+                  onMouseEnter={() => setHoverActive(true)}
+                  onMouseLeave={() => setHoverActive(false)}
+                >
+                  <ArrowDecoration
+                    hide={showing}
+                    active={hoverActive}
+                    src={arrow}
+                  />
+                </ArrowDecorationDiv>
+                <RewardsHeaderContainer
+                  onMouseEnter={() => setHoverActive(true)}
+                  onMouseLeave={() => setHoverActive(false)}
+                >
+                  <RewardsHeader>
+                    Redeem rewards only will be available on
+                    <RewardsDate>XX/XX/XXXX</RewardsDate>
+                  </RewardsHeader>
+                </RewardsHeaderContainer>
+                <RewardsButtonContainer
+                  onMouseEnter={() => setHoverActive(true)}
+                  onMouseLeave={() => setHoverActive(false)}
+                >
+                  <RewardsButtonInner
+                    onMouseEnter={() => setHoverActive(true)}
+                    onMouseLeave={() => setHoverActive(false)}
```

Comment:
Is it needed to have this logic on every element? I think in theory it should work if we just have it on the highest level element. This could cause some bugs having it everywhere I expect, as the cursor could still be in the surrounding area, but the hover could be turned off because the cursor leaves one of the child elements. Or, if we're just using this for styling, we should probably just use the `:hover` CSS selector and remove all this javascript.